node/cn/peer_set.go

```diff
@@ -440,9 +440,12 @@ func (peers *peerSet) SampleResendPeersByType(nodeType common.ConnType) []Peer {
 	var sampledPeers []Peer
 	switch nodeType {
 	case common.ENDPOINTNODE:
-		sampledPeers = peers.typePeers(common.PROXYNODE)
-		if len(sampledPeers) == 0 {
-			sampledPeers = peers.typePeers(common.ENDPOINTNODE)
+		sampledPeers = peers.typePeers(common.CONSENSUSNODE)
+		if len(sampledPeers) < 2 {
+			sampledPeers = append(sampledPeers, samplingPeers(peers.typePeers(common.PROXYNODE), 2-len(sampledPeers))...)
+		}
+		if len(sampledPeers) < 2 {
```

Comment:
Sampling will need to take into account the priorities? If starting with sampling PN, it seems likely that the PN will be excluded from the sample because of the ENs below.

Reply:
If there are more than 2 PN, we don't add EN anymore. So PN has more priority than EN.